cmake/kokkos_enable_options.cmake

```diff
@@ -48,6 +48,7 @@ KOKKOS_ENABLE_OPTION(COMPILER_WARNINGS    OFF "Whether to print all compiler war
 KOKKOS_ENABLE_OPTION(PROFILING_LOAD_PRINT OFF "Whether to print information about which profiling tools got loaded")
 KOKKOS_ENABLE_OPTION(TUNING               OFF "Whether to create bindings for tuning tools")
 KOKKOS_ENABLE_OPTION(AGGRESSIVE_VECTORIZATION OFF "Whether to aggressively vectorize loops")
+KOKKOS_ENABLE_OPTION(LAUNCH_COMPILER      ON  "Whether to potentially use the launch compiler")
```

Comment:
Another option might be:

`GLOBAL_SET(KOKKOS_LINK_OPTIONS $<IF:$<BOOL:Kokkos_LAUNCH_COMPILER>,-DKOKKOS_DEPENDENCE,>)` based on this setting so you can defer adding in `-DKOKKOS_DEPENDENCE` until we know whether or not we are even using the compiler launcher.

Reply:
This option is ever used as far as I can tell

Reply:
https://github.com/kokkos/kokkos/pull/3601#pullrequestreview-548240349